performance/memory-optimize-udtt.sql

```diff
@@ -68,10 +68,12 @@ declare @tt table(
 	table_type_object_id int not null
 )
 
-insert @tt(schema_id, name, user_type_id, table_type_object_id)
-select schema_id, name, user_type_id, type_table_object_id
-from sys.table_types
-where is_user_defined=1 and is_memory_optimized=0
+insert @tt(schema_id, t.name, user_type_id, table_type_object_id)
+select schema_id, t.name, user_type_id, type_table_object_id
+from sys.table_types t
+inner join sys.indexes i
+on t.type_table_object_id=i.object_id
+where is_user_defined=1 and is_memory_optimized=0 and i.name is not null

```

Comment:
Do `where exists (select * from sys.indexes where ...)` instead of `inner join sys.indexes on ...` - there could be multiple indexes defined for a given type, so `join`ing them would produce duplicate output rows.

Reply:
`i.name is not null` just looks for any named index (not a Heap).  Shouldn't we look for existence of `i.is_primary_key=1`?

Reply:
This script recreates all indexes when memory-optimizing a UDTT (which requires at least one nonclustered index), and primary key is only required for memory optimized types with `SCHEMA_AND_DATA` durability, which we don't use.
https://docs.microsoft.com/en-us/sql/relational-databases/in-memory-oltp/indexes-for-memory-optimized-tables

Reply:
It looks like the sql generated for an item with just a simple index (non-primary key) is still incorrect as well. I'll see if I can fix that too.